src/main/java/mate/academy/Main.java

```diff
@@ -1,7 +1,90 @@
 package mate.academy;
 
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Injector;
+import mate.academy.model.Book;
+
 public class Main {
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private static final Logger logger = Logger.getLogger(Main.class.getName());
+    private static final Long BOOK_ID = 3L;
+
     public static void main(String[] args) {
+        BookDao bookDao = (BookDao) injector.getInstance(BookDao.class);
+        List<Book> books = createSampleBooks();
+
+        //Add books to the database.
+        for (Book book : books) {
+            try {
+                bookDao.create(book);
+            } catch (DataProcessingException e) {
+                logger.severe("DataProcessingException occurred while creating a book: "
+                        + book + " - " + e.getMessage());
+            } catch (Exception e) {
+                logger.severe("Unexpected error occurred while creating book: "
+                        + book + " - " + e.getMessage());
+            }
+        }
+        //Find a book by ID.
+        try {
+            Optional<Book> daoById = bookDao.findById(BOOK_ID);
+            logger.info("Found book by ID: " + daoById);
```

Comment:
When logging the result of `bookDao.findById(BOOK_ID)`, you should consider checking if the `Optional<Book>` is present before logging. This will prevent logging a potentially empty `Optional`. You can use `daoById.ifPresent(book -> logger.info("Found book by ID: " + book));` or handle the absence case appropriately.